llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1103,6 +1104,16 @@ void AsmPrinter::emitBBAddrMapSection(const MachineFunction &MF) {
     // always be computed from their offsets.
     emitLabelDifferenceAsULEB128(MBB.getEndSymbol(), MBBSymbol);
     OutStreamer->emitULEB128IntValue(getBBAddrMapMetadata(MBB));
+    // Emit the index of the corresponding LLVMIR basic block.
+    size_t i = 0;
+    for (auto It = F.begin(); It != F.end(); It++) {
```

Comment:
I think this can be:

```c++
for (size_t i = 0, auto It = F.begin; It != F.end(); It++) { ... }
```

meaning you can also remove the `i++`.

Reply:
Not sure this works if the two variables have a different type. [There's a way](https://stackoverflow.com/questions/2687392/is-it-possible-to-declare-two-variables-of-different-types-in-a-for-loop) but it makes it less readable IMO.

Reply:
I did not know that!

Reply:
Me neither!